libsoftwarecontainer/src/gateway/networkgateway.cpp

```diff
@@ -142,71 +144,52 @@ ReturnCode NetworkGateway::parseRule(const json_t *element, std::vector<Rule> &r
     return ReturnCode::SUCCESS;
 }
 
-ReturnCode NetworkGateway::parsePort(const json_t *element, std::vector<unsigned int> &ports)
+ReturnCode NetworkGateway::parsePort(const json_t *element,  IPTableEntry::portFilter &ports)
 {
-    // Port formatted as single integer
+    // Port formatted as single integer means there is only one port
     if (json_is_integer(element)) {
-        int port = json_integer_value(element);
-        ports.push_back(port);
-
+        auto port = json_integer_value(element);
+        ports.any = true;
+        ports.multiport = false;
+        ports.ports = std::to_string(port);
     // Port formatted as a string representing a range
     } else if (json_is_string(element)) {
-        std::string portRange = json_string_value(element);
-
-        const std::string::size_type n = portRange.find("-");
-        const std::string first = portRange.substr(0, n);
-        const std::string last = portRange.substr(n + 1);
-
-        int startPort;
-        if (!parseInt(first.c_str(), &startPort)) {
-             log_error() << "Starting port in range " << portRange << "is not an integer.";
-             return ReturnCode::FAILURE;
-        }
-
-        int endPort;
-        if (!parseInt(first.c_str(), &endPort)) {
-             log_error() << "End port in range " << portRange << "is not an integer.";
-             return ReturnCode::FAILURE;
-        }
-
-        for (int i = startPort; i <= endPort; ++i) {
-            ports.push_back(i);
-        }
-
+        auto portRange = json_string_value(element);
+        ports.any = true;
+        ports.multiport = true;
+        ports.ports = portRange;
     // Port formatted as a list of integers
     } else if (json_is_array(element)) {
         size_t ix;
         json_t *val;
+        std::string portList = "";
+
         json_array_foreach(element, ix, val) {
             if (!json_is_integer(val)) {
                 log_error() << "Entry in port array is not an integer.";
                 return ReturnCode::FAILURE;
             }
 
-            int port = json_integer_value(element);
-            ports.push_back(port);
+            int port = json_integer_value(val);
+            portList = portList + std::to_string(port) + ",";
```

Comment:
Since it look quite easy to read and also there is + operator programmed for concatenation, it sounds reasonable to me. append method can be used as well but i prefer this one unless there is no objection.